src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsExpensesFromPage.tsx

```diff
@@ -125,15 +230,79 @@ function WorkspaceWorkflowsApprovalsExpensesFromPage({policy, isLoadingReportDat
     }, [isInitialCreationFlow, route.params.policyID, firstApprover, approvalWorkflow?.action]);
 
     const nextStep = useCallback(() => {
-        const members: Member[] = selectedMembers.map((member) => ({displayName: member.text, avatar: member.icons.at(0)?.source, email: member.login}));
+        // Separate members into existing members and non-members (users to invite)
+        const existingMembers: Member[] = [];
+        const usersToInvite: Array<{email: string; accountID?: number}> = [];
+
+        for (const member of selectedMembers) {
+            const isPolicyMember = policy?.employeeList?.[member.login];
+            if (isPolicyMember) {
+                existingMembers.push({displayName: member.text, avatar: member.icons.at(0)?.source, email: member.login});
+            } else {
+                const iconId = member.icons.at(0)?.id;
+                const accountID = typeof iconId === 'number' ? iconId : undefined;
+                usersToInvite.push({
+                    email: member.login,
+                    accountID,
+                });
+            }
+        }
+
+        // If there are users to invite, navigate to invite flow
+        if (usersToInvite.length > 0) {
+            const invitedEmailsToAccountIDs: Record<string, number> = {};
+            for (const user of usersToInvite) {
+                if (user.email && user.accountID) {
+                    invitedEmailsToAccountIDs[user.email] = user.accountID;
+                } else if (user.email) {
+                    // If no accountID, use default (for new users)
+                    invitedEmailsToAccountIDs[user.email] = CONST.DEFAULT_NUMBER_ID;
+                }
+            }
+
+            setWorkspaceInviteMembersDraft(route.params.policyID, invitedEmailsToAccountIDs);
+
+            // Store the existing members and users to invite in the workflow for after invite
+            // Ensure avatars are only strings (URLs) or undefined, never React components
+            const allMembers: Member[] = [
+                ...existingMembers.map((member) => ({
+                    ...member,
+                    avatar: typeof member.avatar === 'string' ? member.avatar : undefined,
+                })),
+                ...usersToInvite.map((user) => ({
+                    displayName: user.email,
+                    email: user.email,
+                    avatar: undefined,
+                })),
+            ];
+            setApprovalWorkflowMembers(allMembers);
```

Comment:
Do you think it's same with https://github.com/Expensify/App/pull/75589/files#diff-0a4739399b93f09a6f9466afba489945b2c9dc18700f579b75e71a0d6e9167ffR290-R298

If yes, we can move it to the top to apply for both `usersToInvite.length > 0` and else